python/engine/loop.py

```
@@ -517,16 +575,55 @@ def run_build(bcfg: BuildConfig):
+                    from engine.generation import generate_learning
+                    for prompt_text in SAMPLE_PROMPTS:
+                        full_restore(gpu_model, ckpt_snapshot)
+                        gpu_model.reset_optimizer()
+                        gpu_model.reset_context()
+                        prompt_ids = tokenizer.encode(prompt_text)
+                        tokens, losses, _ = generate_learning(
+                            gpu_model, cfg, prompt_ids,
+                            max_tokens=128, temperature=0.7, lr=bcfg.lr)
+                        gen_text = tokenizer.decode(tokens[len(prompt_ids):])
+                        preview = gen_text[:80].replace("\n", " ")
+                        valid = [l for l in losses if not math.isnan(l)]
```

Ruff (0.15.2)

[error] Ambiguous variable name: `l`

(E741)